apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -22,38 +22,33 @@ class Block_Patterns_From_API {
 	private static $instance = null;
 
 	/**
-	 * Cache key for patterns array.
+	 * Valid source strings for retrieving patterns.
 	 *
-	 * @var string
+	 * @var array
 	 */
-	private $patterns_cache_key;
+	private $valid_patterns_sources = array( 'block_patterns', 'fse_block_patterns' );
 
 	/**
 	 * Block_Patterns constructor.
+	 *
+	 * @param array $patterns_sources A array of strings, each of which matches a valid source for retrieving patterns.
 	 */
-	private function __construct() {
-		$this->patterns_cache_key = sha1(
-			implode(
-				'_',
-				array(
-					'block_patterns',
-					A8C_ETK_PLUGIN_VERSION,
-					$this->get_block_patterns_locale(),
-				)
-			)
-		);
+	private function __construct( $patterns_sources ) {
```

Comment:
Just adding a note that because of the signature changing in the constructor here, to make this safe to deploy non-atomically, we should probably add a default value. That could be a line below this line:

```
$patterns_sources = empty( $patterns_sources ) ? array( 'block_patterns' ) : $patterns_sources;
```

Then, we could move the change to `full-site-editing-plugin.php` out of this PR and into a second follow-up PR to 'switch on' the new behaviour.

Reply:
Good call. Will do 👍